dll/win32/browseui/shellbars/CBandSiteMenu.cpp

```diff
@@ -94,21 +128,213 @@ HRESULT CBandSiteMenu::CreateMenuPart()
         SHGetValue(HKEY_CLASSES_ROOT, wRegKey, NULL, NULL, wszBandName, &dwDataSize);
 
         /* Insert it */
-        InsertMenu(hmenuToolbars, cBands, MF_BYPOSITION, DSA_GetItemCount(m_menuDsa), wszBandName);
-        DSA_AppendItem(m_menuDsa, &iter);
+        InsertMenu(hmenuToolbars, cBands, MF_BYPOSITION, DSA_GetItemCount(m_comcatDsa) + FIRST_COMCAT_MENU_ID, wszBandName);
+        DSA_AppendItem(m_comcatDsa, &iter);
         cBands++;
     }
     while (dwRead > 0);
 
     return S_OK;
 }
 
+HRESULT CBandSiteMenu::_CreateNewISFBand(HWND hwnd, REFIID riid, void** ppv)
+{
+    WCHAR path[MAX_PATH];
+    WCHAR message[256];
+    BROWSEINFOW bi = { hwnd, NULL, path };
+
+    if (LoadStringW(GetModuleHandleW(L"browseui.dll"), IDS_BROWSEFORNEWTOOLAR, message, _countof(message)))
```

Comment:
CAtlString has a LoadString function that handles allocation, and saves you from using fixed buffers.

Reply:
I tried to do it but got linker issues. I'm skipping it for now.